### ISSUE TITLE
[transforms] Add geometric transformations (train detection models with rotated samples)

### ISSUE THREAD

**Issue by charlesmindee** [mindee/doctr#352]
We need to train our detection models with rotated samples, because for now we support rotation but our models are not fitted to predict rotated boxes, which yields to very poor results.

**Comment by Ismail-Maj** [comment]
Hello,
I don't know if you're open to outsider contribution, but I think there might be a bug in your function (I don't know enough about the flow of the program to tell), line 46 doctr/transforms/functional/pytorch.py.

It's related to this problem: https://www.pyimagesearch.com/2017/01/02/rotate-images-correctly-with-opencv-and-python/
I had the same problem when training my OCR on rotated images, some part of the text went out of bound.

If it is in fact a problem, you could replace rotate from torchvision to imutils.rotate_bound, it's not ideal (extra import, tensor to array) so I think the best way to do it is to calculate the needed padding, pad the image then rotate with torchvision.

**Comment by fg-mindee** [comment]
Hi @Ismail-Maj :wave: 

Thanks for reporting this! I'll take a look at it indeed, fortunately for now we haven't used rotation in our training (only as a preprocessing step in some cases) so there aren't any major consequences on existing checkpoints.

But this could get messy later on so thanks for this :pray: 
We certainly do accept outsider contributors, this is an open-source project, feel free to open issues/PRs :)

**Comment by charlesmindee** [comment]
![Screenshot_2021-09-16 app · Streamlit(1)](https://user-images.githubusercontent.com/70526046/133616968-3d0b20d8-aa64-4b92-a021-d5f182cd02b7.png)
![Screenshot_2021-09-16 app · Streamlit](https://user-images.githubusercontent.com/70526046/133616985-35a965c1-c153-495a-af18-fd73b95b5da4.png)

**Comment by fg-mindee** [comment]
Closed by #765 :rocket:
